interdiff/agent.py

```diff
@@ -1,65 +1,59 @@
-    log_render: bool = struct.field(pytree_node=False, default=False)
+import os
 
 
-class Agent(struct.PyTreeNode):
-    hparams: HParams
+class RLAgent:
+    def __init__(self, model, optimizer, hparams: Dict[str, Any]):
+        self.model = model
+        self.optimizer = optimizer
+        self.log_frequency = hparams.get("log_frequency", 1)
+        self.log_to_wandb = hparams.get("log_to_wandb", False)
+        self.step = 0
+        self.hparams = hparams
+        self.save_dir = self.hparams['save_dir']
+        self.save_frequency = hparams.get("save_frequency", None)
 
-    def train(self, rng: jax.Array) -> Tuple[TrainState, Dict[str, jax.Array]]:
+    def train_step(self, batch: Dict[str, Any]) -> Dict[str, float]:
         raise NotImplementedError
 
-    def log(self, logs, inspectable=None):
-        if len(logs) == 0 or logs["iter/updates"] % self.hparams.log_frequency != 0:
-            return
-
-        start_time = time.time()
-        msg = f"Update Step: {logs['iter/updates']}, Frames: {logs['iter/frames']}"
-        step = jnp.asarray(logs["iter/updates"], dtype=jnp.int32)
-
-        # log renders
-        if self.hparams.log_render:
-            render_human = logs.pop("render/human")  # (T, 3, H, W)
-            logs[f"render/human"] = wandb.Video(np.array(render_human), fps=4)
-
-        if "done_mask" in logs:
-            mask = jnp.asarray(logs.pop("done_mask"), dtype=jnp.bool)  # (T, N)
-            # log episode length
-            if "lengths" in logs:
-                lengths: jax.Array = logs.pop("lengths")  # (T, N)
-                episode_lengths = lengths[mask]  # (K,)
-                logs["perf/episode_length"] = jnp.mean(episode_lengths)
-                msg += f", Length: {logs['perf/episode_length']}"
-
-            # log returns
-            if "returns" in logs:
-                returns = logs.pop("returns")  # (T, N)
-                final_returns = returns[mask]  # (K,)
-                logs["perf/returns"] = jnp.mean(final_returns)
-                logs["perf/success_rate"] = jnp.mean(final_returns == 1.0)
-                msg += f", Returns: {logs['perf/returns']}, Success Rate: {logs['perf/success_rate']}"
-
-        msg += f", Logging time cost: {time.time() - start_time}"
-        wandb.log(logs, step=step)
-
-    def log_on_train_end(self, logs):
-        print(jax.tree.map(lambda x: x.shape, logs))
-        len_logs = len(logs["iter/updates"])
-        for step in range(len_logs):
-            step_logs = {k: jax.tree.map(lambda x: x[step], v) for k, v in logs.items()}
-            self.log(step_logs)
\ No newline at end of file
+    def log(self, metrics: Dict[str, float]):
+        if self.step % self.log_frequency == 0:
+            print(f"Step {self.step}: {metrics}")
+            if self.log_to_wandb:
+                wandb.log(metrics, step=self.step)
+
+    def save_model(self):
+
+        if not os.path.isdir(self.save_dir):
+            os.mkdir(self.save_dir)
```

Comment:
```suggestion
            os.makedirs(self.save_dir, exists_ok=True)
```
or it will fail if the folder already exists